src/main/java/ekud/task/Task.java

```diff
@@ -0,0 +1,175 @@
+        isDone = false;
+    }
+
+    /**
+     * Returns the Task corresponding to a given task's save-string. If a given taskSaveString is invalid,
+     * a warning is printed and null is returned.
+     *
+     * @param taskSaveString The task's save-string.
+     * @param ui The ui which prints the warning for an invalid taskSaveString.
+     * @return Task corresponding to taskSaveString.
+     * @see Ui
+     * @see ekud.components.Storage#loadTasks(TaskList, Ui)
+     */
+    public static Task getTaskFromSave(String taskSaveString, Ui ui) {
+        try {
+            taskSaveString = taskSaveString.trim(); // removes extra new lines
+            String[] args = taskSaveString.split("(\\s\\|\\s)"); // splits string along < | > delimiter
+
+            String type = args[0];
+            boolean isDone = args[1].equals("1");
+            // CHECKSTYLE.OFF: Indentation
+            Task task = switch (type) {
+                case "T" -> new TodoTask(args[2]);
+                case "D" -> new DeadlineTask(args[2], args[3]);
+                case "E" -> new EventTask(args[2], args[3], args[4]);
+                default -> null;
+            };
+            // CHECKSTYLE.ON: Indentation
+
+            if (task != null) {
+                task.isDone = isDone;
+            }
+
+            return task;
+
+        } catch (IndexOutOfBoundsException | EkudException e) {
+            String message = String.format(
+                        "Warning: ekud.task.Task entry { %s } is missing required arguments or is"
+                                + "incorrectly formatted\nRemoving ekud.task entry...",
+                        taskSaveString);
+            ui.addToBuffer(message);
+            return null;
+        }
+    }
+
+    /**
+     * Creates a Task based on a given set of tokens.
+     *
+     * @param tokens A {@link HashMap} that maps tokens to {@link String} values.
+     * @return {@link Task} based on tokens
+     * @throws EkudException If the tokens map has missing or invalid tokens.
+     * @see EkudException
+     */
+    public static Task getTaskFromTokens(HashMap<String, String> tokens) throws EkudException {
+        String type = tokens.get("command").toLowerCase();
+        String argument = tokens.get("argument");
+
+        // CHECKSTYLE.OFF: Indentation
+        return switch (type) {
+            case "todo" -> new TodoTask(argument);
+            case "deadline" -> new DeadlineTask(argument, tokens.get("/by"));
+            case "event" -> new EventTask(argument, tokens.get("/from"), tokens.get("/to"));
+            default -> throw new EkudException("Wow! What is this type of ekud.task?"
+                    + "\nI'm not sure how to process this");
+        };
+        // CHECKSTYLE.ON: Indentation
+    }
+
+    /**
+     * Returns an error message {@link String} corresponding to this type of Task.
+     *
+     * @return Error message String.
+     */
+    public abstract String getEmptyDescriptionErrorMessage();
```

Comment:
Good use of abstract method to ensure that the specific tasks have relevant error messages!